tsl/src/nodes/decompress_chunk/compressed_batch.c

```diff
@@ -659,19 +652,20 @@ compressed_batch_advance(DecompressChunkState *chunk_state, DecompressBatchState
 					column_values->iterator->try_next(column_values->iterator);
 				}
 			}
-			InstrCountFiltered1(&chunk_state->csstate, 1);
+
+			InstrCountFiltered1(dcontext->ps, 1);
 			continue;
 		}
 
-		make_next_tuple(chunk_state, batch_state);
+		make_next_tuple(dcontext, batch_state);
 
-		if (!postgres_qual(chunk_state, batch_state))
+		if (!postgres_qual(dcontext, batch_state))
 		{
 			/*
 			 * The tuple didn't pass the qual, fetch the next one in the next
 			 * iteration.
 			 */
-			InstrCountFiltered1(&chunk_state->csstate, 1);
+			InstrCountFiltered1(dcontext->ps, 1);
```

Comment:
Oh, I thought you managed to remove the dependency on the DecompressChunk node, but turns out it even has an additional layer of indirection now :smiling_face_with_tear: What's the plan, copy the quals into the DecompressContext and pass these instrumentations out of it? Anyway, not related to this PR.

Reply:
It now references a generic `PlanState`  node, which could by _any_ scan node and not _specifically_ `DecompressChunkState` like before.

But this is also ongoing refactoring and don't want to do too much in one go.